animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,145 @@
+
+    def getExecutionFunctionName(self):
+        if self.mode == "INDEX":
+            if self.useIndexList:
+                if self.useFloatList:
+                    return "execute_IndicesList"
+                else:
+                    return "execute_IndicesSingle"
+            else:
+                return "execute_Index"
+        elif self.mode == "ALL":
+            if self.useFloatList:
+                return "execute_AllList"
+            else:
+                return "execute_AllSingle"
+
+    def execute_Index(self, object, identifier, index, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        vertexGroup.add([index], weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesSingle(self, object, identifier, indices, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesList(self, object, identifier, indices, weights):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        for i, index in enumerate(indices):
+            vertexGroup.add([index], weights[i], "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_AllSingle(self, object, identifier, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        indices = list(range(0, len(object.data.vertices)))
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_AllList(self, object, identifier, weights):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        totalVertices = len(object.data.vertices)    
+        for i, weight in enumerate(weights):
+            if i >= totalVertices: return 
+            vertexGroup.add([i], weight, "REPLACE") 
```

Comment:
Use a `virtualDoubleList`.

Reply:
To create a virtual list, e.g., `list = virtualDoubleList.create()`, what arguments are required in _create()_? Can you please give an example of how to use _virtualDoubleList_?

Reply:
The `create()` function takes two required arguments. The first argument is a list object and the second argument is a default value.
A virtual list can fill itself with the default value if it is empty. For instance:

```python
v = virtualDoubleList.create([], 5)
print(v[0]) # Will print 5
print(v[1]) # Will print 5
...
```
A virtual list can also repeat itself when you access elements that are out of bound. For instance:

```python
v = virtualDoubleList.create([0, 2], 5)
print(v[0]) # Will print 0
print(v[1]) # Will print 2
print(v[2]) # Will print 0
print(v[3]) # Will print 2
...

So a virtual lists is never out of bound, regardless of the length of the actual list.

Reply:
Thanks for the example.